src/schema.rs

```diff
@@ -0,0 +1,75 @@
+//! Schema
+
+#![allow(missing_docs)]
+use template;
+use std::collections::BTreeMap;
+use serde_json::Value;
+
+/// Schema
+pub struct Schema {
+    files: BTreeMap<String, File>,
+}
+
+/// Supported output file formats
+pub enum FileFormat {
+    Ini,
+    Json,
+}
+
+/// File
+pub struct File {
+    /// The file format to generate
+    format: FileFormat,
+    /// Whether to operate in fileset mode
+    fileset: bool,
+    /// The location of the file on the filesystem
+    location: Location,
+    /// The properties defining how this file is transformed
+    properties: Vec<Property>,
+}
+
+/// A partition where a filesystem can be found
+pub enum Partition {
+    /// A primary partition
+    Primary(u64),
+    /// A logical partitoin
+    Logical {
+        /// The primary partition containing the logical partition table
+        on_primary: u64,
```

Comment:
It'd be so nice if we can create a subtype of an unsigned integer type constrained to `0..4`, like Ada :)